coffeespecs.py

```diff
@@ -165,109 +156,182 @@ def fromJSON(coffee_json):
         return coffee
 
 
+def parse_abbreviation(all_abbreviation_tokens_by_spec, token_input, remaining_specs):
+    for spec in remaining_specs:
+        if token_input in all_abbreviation_tokens_by_spec[spec]:
+            return ((spec, token_input), )
+
+    for spec in remaining_specs:
+        for token in all_abbreviation_tokens_by_spec[spec]:
+            if token_input.startswith(token):
+                copy_of_remaining_specs = list(remaining_specs)
+                copy_of_remaining_specs.remove(spec)
+                remainder = token_input[len(token):]
+                remainder_result = parse_abbreviation(all_abbreviation_tokens_by_spec, remainder, copy_of_remaining_specs)
+                if remainder_result is not None:
+                    return ((spec, token),) + remainder_result
+    return None
+
+
+class CoffeeSpecOption(object):
+    def __init__(self, specname, name, abbreviations, word_tokens):
+        self.specname = specname
+        self.name = name
+        self.word_tokens = [x.lower() for x in word_tokens]
```

Comment:
Why are these lists, and not just a set? (since you go to a bunch of effort later to put them in a set)